lodash.js

```diff
@@ -2494,14 +2494,19 @@
      *
      * @private
      * @param {Array} array The array to flatten.
-     * @param {boolean} [isDeep] Specify a deep flatten.
+     * @param {number} [maxDepth=1] The maximum recursion depth.
      * @param {boolean} [isStrict] Restrict flattening to arrays-like objects.
      * @param {Array} [result=[]] The initial result value.
      * @returns {Array} Returns the new flattened array.
      */
-    function baseFlatten(array, isDeep, isStrict, result) {
+    function baseFlatten(array, maxDepth, isStrict, result) {
+      maxDepth = maxDepth === undefined ? 1 : nativeMax(toInteger(maxDepth), 0);
       result || (result = []);
 
+      if (maxDepth < 1) {
```

Comment:
I intentionally made it callable with `maxDepth == 0` for flexibility purposes. It defaults to 1. I'll change it if you think it's necessary.